Source/Synthesis/FieldTypes/DateTimeField.cs

```diff
@@ -21,8 +21,14 @@ public virtual DateTime Value
 			{
 				if (!IsFieldLoaded && InnerSearchValue != null)
 				{
+					DateTime ret;
+					if (DateTime.TryParse(InnerSearchValue, out ret))
+						return ret;
+					if (DateTime.TryParseExact(InnerSearchValue, "yyyyMMdd'T'HHmm'Z'", CultureInfo.CurrentCulture,
+						DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal, out ret))
+						return ret;
 					var converter = new IndexFieldDateTimeValueConverter();
-					// ReSharper disable once PossibleNullReferenceException
+					//// ReSharper disable once PossibleNullReferenceException
 					return (DateTime)converter.ConvertFrom(InnerSearchValue);
```

Comment:
I was having serious issues reliably getting the datetimes to parse from the index.  I believe this is due to sitecore changing dates to UTC.  But i believe these three methods falling back should work for sitecore 7 or 8